### PR TITLE
Draft: Williamhe/1357976 add python run build command

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as \#\<issue\_number\>.
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: src/BuildScriptGenerator/Python/PythonPlatform.cs [hunk 1]
```diff
@@ -162,6 +162,13 @@ public PlatformDetectorResult Detect(RepositoryContext context)
 
             var manifestFileProperties = new Dictionary<string, string>();
 
+            string runBuildCommand = Environment.GetEnvironmentVariable("RUN_BUILD_COMMAND");
+            if (!String.IsNullOrEmpty(runBuildCommand))
+            {
+                _logger.LogWarning("RUN_BUILD_COMMAND environment variable detected");
```
Comment (arroyc) [inline src/BuildScriptGenerator/Python/PythonPlatform.cs:168]: nit: logdebug  would be better IMO

**Review by arroyc — CHANGES_REQUESTED** [review]
rebase to main and then rerun validation build .. because you are using flask  app which broke our pipeline after the dependencies got updated. we want to make sure this test is running fine before we merge them to main

File: src/BuildScriptGenerator/Python/PythonPlatform.cs (hunk 1 above)
Comment (william-msft) [reply src/BuildScriptGenerator/Python/PythonPlatform.cs]: Agree, updated

File: doc/configuration.md [hunk 2]
```diff
@@ -48,6 +48,7 @@ PYTHON\_VERSION              | Specify which Python version the app is using
 DISABLE\_PYTHON\_BUILD       | Do not apply Python build even if repo indicates it            | `false` | `true`, `false`
 VIRTUALENV\_NAME             | Specify Python virtual environment name                        | ""      | "antenv2.7"
 DISABLE\_COLLECTSTATIC       | Disable running `collectstatic` when building Django apps.      | `false` | `true`, `false`
+RUN_BUILD_COMMAND            | Custom command ran at the very end. After virtual envionment, `pip install`, creating python package egg, enabling collecstatic, compresing virtual environment (where applicable)'                                           | ""  | "echo hello world"
```
Comment (cormacpayne) [inline doc/configuration.md:51]: What's the difference between this `RUN_BUILD_COMMAND` and the `POST_BUILD_COMMAND`? Just from reading the descriptions they seem similar, but it's hard to tell the difference.

File: src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl [hunk 3]
```diff
@@ -244,4 +243,10 @@ fi
 			fi
 		fi
 	{{ end }}
-{{ end }}
\ No newline at end of file
+{{ end }}
+
+if [ ! -z "$RUN_BUILD_COMMAND" ]
+then
+	echo 'Running $RUN_BUILD_COMMAND'
```
Comment (cormacpayne) [inline src/BuildScriptGenerator/Python/PythonBashBuildSnippet.sh.tpl:250]: nit: just to make the log a little more clear, I would suggest changing the wording here to:

`echo 'Running the following provided RUN_BUILD_COMMAND: $RUN_BUILD_COMMAND`

File: src/BuildScriptGenerator/Python/PythonPlatform.cs [hunk 4]
```diff
@@ -162,6 +162,13 @@ public PlatformDetectorResult Detect(RepositoryContext context)
 
             var manifestFileProperties = new Dictionary<string, string>();
 
+            string runBuildCommand = Environment.GetEnvironmentVariable("RUN_BUILD_COMMAND");
```
Comment (cormacpayne) [inline src/BuildScriptGenerator/Python/PythonPlatform.cs:165]: nit: I'd suggest pulling the `"RUN_BUILD_COMMAND"` string out into its own private `const` defined at the top of this file.

**Review by cormacpayne — CHANGES_REQUESTED** [review]

**Comment by william-msft** [comment]
Closing until we re-visit

**Pull request closed without merging**